incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -91,13 +91,68 @@ func (r *ConfigReconciler) getSingleton(ctx context.Context) (*api.HNCConfigurat
 		if !errors.IsNotFound(err) {
 			return nil, err
 		}
+	}
 
-		// It doesn't exist - initialize it to a default value.
-		inst.Spec = config.GetDefaultConfigSpec()
+	r.validateSingleton(inst)
+	return inst, nil
+}
+
+// validateSingleton sets the singleton name if it hasn't been set and ensures
+// Role and RoleBinding have the default configuration.
+func (r *ConfigReconciler) validateSingleton(inst *api.HNCConfiguration) {
+	// It is possible that the singleton does not exist on the apiserver. In this
+	// case its name hasn't been set yet.
+	if inst.ObjectMeta.Name == "" {
+		r.Log.Info(fmt.Sprintf("Set the object name to be %s", api.HNCConfigSingleton))
 		inst.ObjectMeta.Name = api.HNCConfigSingleton
 	}
+	r.validateRBACTypes(inst)
+}
 
-	return inst, nil
+// validateRBACTypes set Role and RoleBindings to the default configuration if they are
+// missing or having non-default configuration in the Spec.
+func (r *ConfigReconciler) validateRBACTypes(inst *api.HNCConfiguration) {
+	roleExists, roleBindingsExists := false, false
+
+	// Check the mode of Role and RoleBinding. The mode can be either the default mode
+	// or not set; otherwise, we will change the mode to the default mode.
+	for i := 0; i < len(inst.Spec.Types); i++ {
+		t := &inst.Spec.Types[i]
+		if r.isRole(*t) {
+			mode := config.GetDefaultRoleSpec().Mode
+			if t.Mode != mode && t.Mode != "" {
+				r.Log.Info(fmt.Sprintf("Invalid mode for Role. Change the mode from %s to %s", t.Mode, mode))
```

Comment:
"Changing the mode" is better, because it shows the action that HNC has taken

Reply:
Thanks for the suggestion! Updated "change" along with two other verbs ("add" and "set") in this PR.